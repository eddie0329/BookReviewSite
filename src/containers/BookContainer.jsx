import { connect } from "react-redux";
import {
  recieveBooks,
  deleteBook,
  undoDeleteBook,
  recieveBooksThunk,
  deleteBookThunk,
  addBookThunk,
  recieveBook,
  recieveBookThunk
} from "../actions";
import BookList from "../components/BookList";

const mapStateToProps = state => ({
  books: state.books,
  loading: state.loading,
  book: state.select,
});

const mapDispatchToProps = dispatch => ({
  recieveBooks: books => {
    dispatch(recieveBooks(books));
  },
  deleteBook: bookId => {
    dispatch(deleteBook(bookId));
  },
  undoDeleteBook: bookId => {
    dispatch(undoDeleteBook(bookId));
  },
  recieveBooksThunk: token => {
    dispatch(recieveBooksThunk(token));
  },
  deleteBookThunk: (token, bookId) => {
    dispatch(deleteBookThunk(token, bookId));
  },
  addBookThunk: (token, book) => {
    dispatch(addBookThunk(token, book));
  },
  recieveBook: book => {
    dispatch(recieveBook(book));
  },
  recieveBookThunk: (token, bookId) => {
    dispatch(recieveBookThunk(token, bookId));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BookList);
