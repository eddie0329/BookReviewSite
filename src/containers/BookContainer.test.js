import React from "react";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import BookContainer from "./BookContainer";
Enzyme.configure({ adapter: new Adapter() });

describe("BookContainer", () => {
  it("render sucessfully", () => {
    const mockStore = configureMockStore([thunk]);

    let store = mockStore({
      books: [],
      loading: false
    });
    const component = mount(<BookContainer token={null} store={store} />);
    expect(component).toMatchSnapshot();
  });

  it("including books render sucessfully", () => {
    const mockStore = configureMockStore([thunk]);
    const booksMock = [
      {
        bookId: 1,
        ownerId: "7d26db27-168c-4c6a-bd9a-9e20677b60b8",
        title: "모던 자바스크립트 입문",
        message: "모던하군요",
        author: "서재원",
        url:
          "http://www.kyobobook.co.kr/product/detailViewKor.laf?ejkGb=KOR&mallGb=KOR&barcode=9791160504439&orderClick=LET&Kc=",
        createdAt: "2019-05-15T07:18:14.000Z",
        updatedAt: "2019-06-07T22:56:50.000Z",
        deletedAt: null
      }
    ];
    let store = mockStore({
      books: booksMock,
      loading: false
    });
    const component = mount(<BookContainer token={null} store={store} />);
    expect(component).toMatchSnapshot();
  });
});
