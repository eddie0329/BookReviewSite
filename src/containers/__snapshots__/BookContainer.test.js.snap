// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookContainer including books render sucessfully 1`] = `
<Connect(BookList)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  token={null}
>
  <BookList
    addBookThunk={[Function]}
    books={
      Array [
        Object {
          "author": "서재원",
          "bookId": 1,
          "createdAt": "2019-05-15T07:18:14.000Z",
          "deletedAt": null,
          "message": "모던하군요",
          "ownerId": "7d26db27-168c-4c6a-bd9a-9e20677b60b8",
          "title": "모던 자바스크립트 입문",
          "updatedAt": "2019-06-07T22:56:50.000Z",
          "url": "http://www.kyobobook.co.kr/product/detailViewKor.laf?ejkGb=KOR&mallGb=KOR&barcode=9791160504439&orderClick=LET&Kc=",
        },
      ]
    }
    deleteBook={[Function]}
    deleteBookThunk={[Function]}
    loading={false}
    recieveBooks={[Function]}
    recieveBooksThunk={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    token={null}
    undoDeleteBook={[Function]}
  >
    <Affix
      offsetTop={120}
      onChange={[Function]}
      target={[Function]}
    >
      <ReactResizeObserver
        onResize={[Function]}
      >
        <div>
          <div
            className=""
          >
            <ReactResizeObserver
              onResize={[Function]}
            >
              <Button
                block={false}
                ghost={false}
                htmlType="button"
                loading={false}
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "#b53131",
                    "border": "black",
                    "marginTop": "50px",
                    "position": "absolute",
                    "width": "110px",
                  }
                }
                type="primary"
              >
                <Wave>
                  <button
                    className="ant-btn ant-btn-primary"
                    onClick={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#b53131",
                        "border": "black",
                        "marginTop": "50px",
                        "position": "absolute",
                        "width": "110px",
                      }
                    }
                    type="button"
                  >
                    <Icon
                      key=".0"
                      type="plus-circle"
                    >
                      <LocaleReceiver
                        componentName="Icon"
                      >
                        <i
                          aria-label="icon: plus-circle"
                          className="anticon anticon-plus-circle"
                        >
                          <IconReact
                            className=""
                            type="plus-circle-o"
                          >
                            <svg
                              aria-hidden="true"
                              className=""
                              data-icon="plus-circle"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              key="svg-plus-circle"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                                key="svg-plus-circle-svg-0"
                              />
                              <path
                                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                                key="svg-plus-circle-svg-1"
                              />
                            </svg>
                          </IconReact>
                        </i>
                      </LocaleReceiver>
                    </Icon>
                    <span
                      key=".1"
                    >
                      Add Book
                    </span>
                  </button>
                </Wave>
              </Button>
              <Modal
                centered={true}
                confirmLoading={false}
                footer={null}
                maskTransitionName="fade"
                okType="primary"
                onCancel={[Function]}
                transitionName="zoom"
                visible={false}
                width="600px"
              >
                <Component
                  closeIcon={
                    <span
                      className="ant-modal-close-x"
                    >
                      <Icon
                        className="ant-modal-close-icon"
                        type="close"
                      />
                    </span>
                  }
                  confirmLoading={false}
                  footer={null}
                  maskTransitionName="fade"
                  okType="primary"
                  onCancel={[Function]}
                  onClose={[Function]}
                  prefixCls="ant-modal"
                  transitionName="zoom"
                  visible={false}
                  width="600px"
                  wrapClassName="ant-modal-centered"
                >
                  <PortalWrapper
                    visible={false}
                  />
                </Component>
              </Modal>
            </ReactResizeObserver>
          </div>
        </div>
      </ReactResizeObserver>
    </Affix>
    <div
      style={
        Object {
          "height": "835px",
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": "2%",
        }
      }
    >
      <Books
        author="서재원"
        bookId={1}
        createdAt="2019-05-15T07:18:14.000Z"
        deleteBookThunk={[Function]}
        deletedAt={null}
        key="0"
        message="모던하군요"
        ownerId="7d26db27-168c-4c6a-bd9a-9e20677b60b8"
        title="모던 자바스크립트 입문"
        token={null}
        undoDeleteBook={[Function]}
        updatedAt="2019-06-07T22:56:50.000Z"
        url="http://www.kyobobook.co.kr/product/detailViewKor.laf?ejkGb=KOR&mallGb=KOR&barcode=9791160504439&orderClick=LET&Kc="
      >
        <div
          style={
            Object {
              "background": "#ebe5da",
              "display": "flex",
              "marginLeft": "auto",
              "marginRight": "auto",
              "padding": "30px",
            }
          }
        >
          <Card
            bordered={true}
            extra={
              <Button
                block={false}
                ghost={false}
                htmlType="button"
                loading={false}
                onClick={[Function]}
                size="small"
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                <Icon
                  type="close"
                />
              </Button>
            }
            hoverable={true}
            onClick={[Function]}
            style={
              Object {
                "width": 1000,
              }
            }
            title="모던 자바스크립트 입문"
          >
            <div
              className="ant-card ant-card-bordered ant-card-hoverable"
              onClick={[Function]}
              style={
                Object {
                  "width": 1000,
                }
              }
            >
              <div
                className="ant-card-head"
                style={Object {}}
              >
                <div
                  className="ant-card-head-wrapper"
                >
                  <div
                    className="ant-card-head-title"
                  >
                    모던 자바스크립트 입문
                  </div>
                  <div
                    className="ant-card-extra"
                  >
                    <Button
                      block={false}
                      ghost={false}
                      htmlType="button"
                      loading={false}
                      onClick={[Function]}
                      size="small"
                      style={
                        Object {
                          "float": "right",
                        }
                      }
                    >
                      <Wave>
                        <button
                          className="ant-btn ant-btn-sm"
                          onClick={[Function]}
                          style={
                            Object {
                              "float": "right",
                            }
                          }
                          type="button"
                        >
                          <Icon
                            key=".0"
                            type="close"
                          >
                            <LocaleReceiver
                              componentName="Icon"
                            >
                              <i
                                aria-label="icon: close"
                                className="anticon anticon-close"
                              >
                                <IconReact
                                  className=""
                                  type="close-o"
                                >
                                  <svg
                                    aria-hidden="true"
                                    className=""
                                    data-icon="close"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    key="svg-close"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                      key="svg-close-svg-0"
                                    />
                                  </svg>
                                </IconReact>
                              </i>
                            </LocaleReceiver>
                          </Icon>
                        </button>
                      </Wave>
                    </Button>
                  </div>
                </div>
              </div>
              <div
                className="ant-card-body"
                style={Object {}}
              >
                <p>
                  COMMENT: 
                </p>
                <p>
                  AUTHOR: 
                  서재원
                </p>
                <p>
                  URL: 
                  <a
                    herf="http://www.kyobobook.co.kr/product/detailViewKor.laf?ejkGb=KOR&mallGb=KOR&barcode=9791160504439&orderClick=LET&Kc="
                  >
                    http://www.kyobobook.co.kr/product/detailViewKor.laf?ejkGb=KOR&mallGb=KOR&barcode=9791160504439&orderClick=LET&Kc=
                  </a>
                </p>
              </div>
            </div>
          </Card>
          <Modal
            centered={true}
            confirmLoading={false}
            footer={null}
            maskTransitionName="fade"
            okType="primary"
            onCancel={[Function]}
            title="모던 자바스크립트 입문"
            transitionName="zoom"
            visible={false}
            width={520}
          >
            <Component
              closeIcon={
                <span
                  className="ant-modal-close-x"
                >
                  <Icon
                    className="ant-modal-close-icon"
                    type="close"
                  />
                </span>
              }
              confirmLoading={false}
              footer={null}
              maskTransitionName="fade"
              okType="primary"
              onCancel={[Function]}
              onClose={[Function]}
              prefixCls="ant-modal"
              title="모던 자바스크립트 입문"
              transitionName="zoom"
              visible={false}
              width={520}
              wrapClassName="ant-modal-centered"
            >
              <PortalWrapper
                visible={false}
              />
            </Component>
          </Modal>
        </div>
      </Books>
    </div>
    <Pagination
      defaultCurrent={1}
      defaultPageSize={3}
      onChange={[Function]}
      style={
        Object {
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": "10px",
        }
      }
      total={1}
    >
      <LocaleReceiver
        componentName="Pagination"
        defaultLocale={
          Object {
            "items_per_page": "/ page",
            "jump_to": "Goto",
            "jump_to_confirm": "confirm",
            "next_3": "Next 3 Pages",
            "next_5": "Next 5 Pages",
            "next_page": "Next Page",
            "page": "",
            "prev_3": "Previous 3 Pages",
            "prev_5": "Previous 5 Pages",
            "prev_page": "Previous Page",
          }
        }
      >
        <Pagination
          className=""
          defaultCurrent={1}
          defaultPageSize={3}
          hideOnSinglePage={false}
          itemRender={[Function]}
          jumpNextIcon={
            <a
              className="ant-pagination-item-link"
            >
              <div
                className="ant-pagination-item-container"
              >
                <Icon
                  className="ant-pagination-item-link-icon"
                  type="double-right"
                />
                <span
                  className="ant-pagination-item-ellipsis"
                >
                  •••
                </span>
              </div>
            </a>
          }
          jumpPrevIcon={
            <a
              className="ant-pagination-item-link"
            >
              <div
                className="ant-pagination-item-container"
              >
                <Icon
                  className="ant-pagination-item-link-icon"
                  type="double-left"
                />
                <span
                  className="ant-pagination-item-ellipsis"
                >
                  •••
                </span>
              </div>
            </a>
          }
          locale={
            Object {
              "items_per_page": "/ page",
              "jump_to": "Goto",
              "jump_to_confirm": "confirm",
              "next_3": "Next 3 Pages",
              "next_5": "Next 5 Pages",
              "next_page": "Next Page",
              "page": "",
              "prev_3": "Previous 3 Pages",
              "prev_5": "Previous 5 Pages",
              "prev_page": "Previous Page",
            }
          }
          nextIcon={
            <a
              className="ant-pagination-item-link"
            >
              <Icon
                type="right"
              />
            </a>
          }
          onChange={[Function]}
          onShowSizeChange={[Function]}
          prefixCls="ant-pagination"
          prevIcon={
            <a
              className="ant-pagination-item-link"
            >
              <Icon
                type="left"
              />
            </a>
          }
          selectComponentClass={[Function]}
          selectPrefixCls="ant-select"
          showLessItems={false}
          showPrevNextJumpers={true}
          showQuickJumper={false}
          showSizeChanger={false}
          showTitle={true}
          style={
            Object {
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": "10px",
            }
          }
          total={1}
        >
          <ul
            className="ant-pagination"
            style={
              Object {
                "marginLeft": "auto",
                "marginRight": "auto",
                "marginTop": "10px",
              }
            }
            unselectable="unselectable"
          >
            <li
              aria-disabled={true}
              className="ant-pagination-disabled ant-pagination-prev"
              onClick={[Function]}
              onKeyPress={[Function]}
              tabIndex={null}
              title="Previous Page"
            >
              <a
                className="ant-pagination-item-link"
              >
                <Icon
                  type="left"
                >
                  <LocaleReceiver
                    componentName="Icon"
                  >
                    <i
                      aria-label="icon: left"
                      className="anticon anticon-left"
                    >
                      <IconReact
                        className=""
                        type="left-o"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="left"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          key="svg-left"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                            key="svg-left-svg-0"
                          />
                        </svg>
                      </IconReact>
                    </i>
                  </LocaleReceiver>
                </Icon>
              </a>
            </li>
            <Pager
              active={true}
              itemRender={[Function]}
              key="1"
              locale={
                Object {
                  "items_per_page": "/ page",
                  "jump_to": "Goto",
                  "jump_to_confirm": "confirm",
                  "next_3": "Next 3 Pages",
                  "next_5": "Next 5 Pages",
                  "next_page": "Next Page",
                  "page": "",
                  "prev_3": "Previous 3 Pages",
                  "prev_5": "Previous 5 Pages",
                  "prev_page": "Previous Page",
                }
              }
              onClick={[Function]}
              onKeyPress={[Function]}
              page={1}
              rootPrefixCls="ant-pagination"
              showTitle={true}
            >
              <li
                className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                onClick={[Function]}
                onKeyPress={[Function]}
                tabIndex="0"
                title={1}
              >
                <a>
                  1
                </a>
              </li>
            </Pager>
            <li
              aria-disabled={true}
              className="ant-pagination-disabled ant-pagination-next"
              onClick={[Function]}
              onKeyPress={[Function]}
              tabIndex={null}
              title="Next Page"
            >
              <a
                className="ant-pagination-item-link"
              >
                <Icon
                  type="right"
                >
                  <LocaleReceiver
                    componentName="Icon"
                  >
                    <i
                      aria-label="icon: right"
                      className="anticon anticon-right"
                    >
                      <IconReact
                        className=""
                        type="right-o"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="right"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          key="svg-right"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                            key="svg-right-svg-0"
                          />
                        </svg>
                      </IconReact>
                    </i>
                  </LocaleReceiver>
                </Icon>
              </a>
            </li>
            <Options
              changeSize={null}
              current={1}
              goButton={false}
              locale={
                Object {
                  "items_per_page": "/ page",
                  "jump_to": "Goto",
                  "jump_to_confirm": "confirm",
                  "next_3": "Next 3 Pages",
                  "next_5": "Next 5 Pages",
                  "next_page": "Next Page",
                  "page": "",
                  "prev_3": "Previous 3 Pages",
                  "prev_5": "Previous 5 Pages",
                  "prev_page": "Previous Page",
                }
              }
              pageSize={3}
              pageSizeOptions={
                Array [
                  "10",
                  "20",
                  "30",
                  "40",
                ]
              }
              quickGo={null}
              rootPrefixCls="ant-pagination"
              selectComponentClass={[Function]}
              selectPrefixCls="ant-select"
            />
          </ul>
        </Pagination>
      </LocaleReceiver>
    </Pagination>
  </BookList>
</Connect(BookList)>
`;

exports[`BookContainer render sucessfully 1`] = `
<Connect(BookList)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  token={null}
>
  <BookList
    addBookThunk={[Function]}
    books={Array []}
    deleteBook={[Function]}
    deleteBookThunk={[Function]}
    loading={false}
    recieveBooks={[Function]}
    recieveBooksThunk={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    token={null}
    undoDeleteBook={[Function]}
  >
    <Affix
      offsetTop={120}
      onChange={[Function]}
      target={[Function]}
    >
      <ReactResizeObserver
        onResize={[Function]}
      >
        <div>
          <div
            className=""
          >
            <ReactResizeObserver
              onResize={[Function]}
            >
              <Button
                block={false}
                ghost={false}
                htmlType="button"
                loading={false}
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "#b53131",
                    "border": "black",
                    "marginTop": "50px",
                    "position": "absolute",
                    "width": "110px",
                  }
                }
                type="primary"
              >
                <Wave>
                  <button
                    className="ant-btn ant-btn-primary"
                    onClick={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#b53131",
                        "border": "black",
                        "marginTop": "50px",
                        "position": "absolute",
                        "width": "110px",
                      }
                    }
                    type="button"
                  >
                    <Icon
                      key=".0"
                      type="plus-circle"
                    >
                      <LocaleReceiver
                        componentName="Icon"
                      >
                        <i
                          aria-label="icon: plus-circle"
                          className="anticon anticon-plus-circle"
                        >
                          <IconReact
                            className=""
                            type="plus-circle-o"
                          >
                            <svg
                              aria-hidden="true"
                              className=""
                              data-icon="plus-circle"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              key="svg-plus-circle"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                                key="svg-plus-circle-svg-0"
                              />
                              <path
                                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                                key="svg-plus-circle-svg-1"
                              />
                            </svg>
                          </IconReact>
                        </i>
                      </LocaleReceiver>
                    </Icon>
                    <span
                      key=".1"
                    >
                      Add Book
                    </span>
                  </button>
                </Wave>
              </Button>
              <Modal
                centered={true}
                confirmLoading={false}
                footer={null}
                maskTransitionName="fade"
                okType="primary"
                onCancel={[Function]}
                transitionName="zoom"
                visible={false}
                width="600px"
              >
                <Component
                  closeIcon={
                    <span
                      className="ant-modal-close-x"
                    >
                      <Icon
                        className="ant-modal-close-icon"
                        type="close"
                      />
                    </span>
                  }
                  confirmLoading={false}
                  footer={null}
                  maskTransitionName="fade"
                  okType="primary"
                  onCancel={[Function]}
                  onClose={[Function]}
                  prefixCls="ant-modal"
                  transitionName="zoom"
                  visible={false}
                  width="600px"
                  wrapClassName="ant-modal-centered"
                >
                  <PortalWrapper
                    visible={false}
                  />
                </Component>
              </Modal>
            </ReactResizeObserver>
          </div>
        </div>
      </ReactResizeObserver>
    </Affix>
    <div
      style={
        Object {
          "height": "835px",
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": "2%",
        }
      }
    />
    <Pagination
      defaultCurrent={1}
      defaultPageSize={3}
      onChange={[Function]}
      style={
        Object {
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": "10px",
        }
      }
      total={0}
    >
      <LocaleReceiver
        componentName="Pagination"
        defaultLocale={
          Object {
            "items_per_page": "/ page",
            "jump_to": "Goto",
            "jump_to_confirm": "confirm",
            "next_3": "Next 3 Pages",
            "next_5": "Next 5 Pages",
            "next_page": "Next Page",
            "page": "",
            "prev_3": "Previous 3 Pages",
            "prev_5": "Previous 5 Pages",
            "prev_page": "Previous Page",
          }
        }
      >
        <Pagination
          className=""
          defaultCurrent={1}
          defaultPageSize={3}
          hideOnSinglePage={false}
          itemRender={[Function]}
          jumpNextIcon={
            <a
              className="ant-pagination-item-link"
            >
              <div
                className="ant-pagination-item-container"
              >
                <Icon
                  className="ant-pagination-item-link-icon"
                  type="double-right"
                />
                <span
                  className="ant-pagination-item-ellipsis"
                >
                  •••
                </span>
              </div>
            </a>
          }
          jumpPrevIcon={
            <a
              className="ant-pagination-item-link"
            >
              <div
                className="ant-pagination-item-container"
              >
                <Icon
                  className="ant-pagination-item-link-icon"
                  type="double-left"
                />
                <span
                  className="ant-pagination-item-ellipsis"
                >
                  •••
                </span>
              </div>
            </a>
          }
          locale={
            Object {
              "items_per_page": "/ page",
              "jump_to": "Goto",
              "jump_to_confirm": "confirm",
              "next_3": "Next 3 Pages",
              "next_5": "Next 5 Pages",
              "next_page": "Next Page",
              "page": "",
              "prev_3": "Previous 3 Pages",
              "prev_5": "Previous 5 Pages",
              "prev_page": "Previous Page",
            }
          }
          nextIcon={
            <a
              className="ant-pagination-item-link"
            >
              <Icon
                type="right"
              />
            </a>
          }
          onChange={[Function]}
          onShowSizeChange={[Function]}
          prefixCls="ant-pagination"
          prevIcon={
            <a
              className="ant-pagination-item-link"
            >
              <Icon
                type="left"
              />
            </a>
          }
          selectComponentClass={[Function]}
          selectPrefixCls="ant-select"
          showLessItems={false}
          showPrevNextJumpers={true}
          showQuickJumper={false}
          showSizeChanger={false}
          showTitle={true}
          style={
            Object {
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": "10px",
            }
          }
          total={0}
        >
          <ul
            className="ant-pagination"
            style={
              Object {
                "marginLeft": "auto",
                "marginRight": "auto",
                "marginTop": "10px",
              }
            }
            unselectable="unselectable"
          >
            <li
              aria-disabled={true}
              className="ant-pagination-disabled ant-pagination-prev"
              onClick={[Function]}
              onKeyPress={[Function]}
              tabIndex={null}
              title="Previous Page"
            >
              <a
                className="ant-pagination-item-link"
              >
                <Icon
                  type="left"
                >
                  <LocaleReceiver
                    componentName="Icon"
                  >
                    <i
                      aria-label="icon: left"
                      className="anticon anticon-left"
                    >
                      <IconReact
                        className=""
                        type="left-o"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="left"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          key="svg-left"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                            key="svg-left-svg-0"
                          />
                        </svg>
                      </IconReact>
                    </i>
                  </LocaleReceiver>
                </Icon>
              </a>
            </li>
            <Pager
              className="ant-pagination-disabled"
              itemRender={[Function]}
              key="noPager"
              locale={
                Object {
                  "items_per_page": "/ page",
                  "jump_to": "Goto",
                  "jump_to_confirm": "confirm",
                  "next_3": "Next 3 Pages",
                  "next_5": "Next 5 Pages",
                  "next_page": "Next Page",
                  "page": "",
                  "prev_3": "Previous 3 Pages",
                  "prev_5": "Previous 5 Pages",
                  "prev_page": "Previous Page",
                }
              }
              onClick={[Function]}
              onKeyPress={[Function]}
              page={0}
              rootPrefixCls="ant-pagination"
              showTitle={true}
            >
              <li
                className="ant-pagination-item ant-pagination-item-0 ant-pagination-disabled ant-pagination-item-disabled"
                onClick={[Function]}
                onKeyPress={[Function]}
                tabIndex="0"
                title={0}
              >
                <a>
                  0
                </a>
              </li>
            </Pager>
            <li
              aria-disabled={true}
              className="ant-pagination-disabled ant-pagination-next"
              onClick={[Function]}
              onKeyPress={[Function]}
              tabIndex={null}
              title="Next Page"
            >
              <a
                className="ant-pagination-item-link"
              >
                <Icon
                  type="right"
                >
                  <LocaleReceiver
                    componentName="Icon"
                  >
                    <i
                      aria-label="icon: right"
                      className="anticon anticon-right"
                    >
                      <IconReact
                        className=""
                        type="right-o"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="right"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          key="svg-right"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                            key="svg-right-svg-0"
                          />
                        </svg>
                      </IconReact>
                    </i>
                  </LocaleReceiver>
                </Icon>
              </a>
            </li>
            <Options
              changeSize={null}
              current={1}
              goButton={false}
              locale={
                Object {
                  "items_per_page": "/ page",
                  "jump_to": "Goto",
                  "jump_to_confirm": "confirm",
                  "next_3": "Next 3 Pages",
                  "next_5": "Next 5 Pages",
                  "next_page": "Next Page",
                  "page": "",
                  "prev_3": "Previous 3 Pages",
                  "prev_5": "Previous 5 Pages",
                  "prev_page": "Previous Page",
                }
              }
              pageSize={3}
              pageSizeOptions={
                Array [
                  "10",
                  "20",
                  "30",
                  "40",
                ]
              }
              quickGo={null}
              rootPrefixCls="ant-pagination"
              selectComponentClass={[Function]}
              selectPrefixCls="ant-select"
            />
          </ul>
        </Pagination>
      </LocaleReceiver>
    </Pagination>
  </BookList>
</Connect(BookList)>
`;
